diff --git a/lib/mspec/helpers/ruby_exe.rb b/lib/mspec/helpers/ruby_exe.rb
index c803ddc..8e9ef94 100644
--- a/lib/mspec/helpers/ruby_exe.rb
+++ b/lib/mspec/helpers/ruby_exe.rb
@@ -91,14 +91,14 @@ class Object
 
   def resolve_ruby_exe
     [:env, :engine, :name, :install_name].each do |option|
-      exe = ruby_exe_options option
+      return unless cmd = ruby_exe_options(option)
+      exe = cmd.split.first
 
       # It has been reported that File.executable is not reliable
       # on Windows platforms (see commit 56bc555c). So, we check the
       # platform. 
-      if exe and File.exists?(exe) and
-          (SpecGuard.windows? || File.executable?(exe))
-        return exe
+      if File.exists?(exe) and (SpecGuard.windows? || File.executable?(exe))
+        return cmd
       end
     end
     nil
